venona/pkg/queue/queue.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 The Codefresh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package queue
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/codefresh-io/go/venona/pkg/logger"
+	"github.com/codefresh-io/go/venona/pkg/monitoring"
+	"github.com/codefresh-io/go/venona/pkg/runtime"
+	"github.com/codefresh-io/go/venona/pkg/task"
+	"github.com/codefresh-io/go/venona/pkg/workflow"
+)
+
+type (
+	// WorkflowQueue manages a map of workflow (id) -> workflow
+	WorkflowQueue struct {
+		runtimes    map[string]runtime.Runtime
+		log         logger.Logger
+		wg          *sync.WaitGroup
+		monitor     monitoring.Monitor
+		queue       chan *workflow.Workflow
+		concurrency int
+		stop        []chan bool
+	}
+)
+
+const defaultWfTaskBufferSize = 20
+
+var (
+	errRuntimeNotFound = errors.New("Runtime environment not found")
+)
+
+// New creates a new TaskQueue instance
+func New(runtimes map[string]runtime.Runtime, log logger.Logger, wg *sync.WaitGroup, monitor monitoring.Monitor, concurrency int) *WorkflowQueue {
+	return &WorkflowQueue{
+		runtimes:    runtimes,
+		log:         log,
+		wg:          wg,
+		monitor:     monitor,
+		queue:       make(chan *workflow.Workflow, 1000),
```

Comment:
Let's save it in a variable

Reply:
i exposed it as a cli flag (+ env variable `WORKFLOW_BUFFER_SIZE`), which defaults to 1000.
i hope that is what you meant.